From 0d1b8e77bec6cb36c410b95779ca2595325bf76f Mon Sep 17 00:00:00 2001
From: "alpha.lin" <alpha.lin@rock-chips.com>
Date: Tue, 29 Mar 2016 16:12:28 +0800
Subject: [PATCH] ARM64: dts: rk3399: Add vpu and rkvdec resource node

Add vpu and rkvdec resource node to enable video codec
supporting for rk3399 android.

Change-Id: I1689955858355b6061957dc43eea17f9b8d71096
Signed-off-by: alpha.lin <alpha.lin@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-fb.dtsi   | 52 +++++++++++++++++++
 .../arm64/boot/dts/rockchip/rk3399-monkey.dts | 15 ++++++
 2 files changed, 67 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-fb.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-fb.dtsi
index a99b02199d21..d37fe5f53fee 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-fb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-fb.dtsi
@@ -42,6 +42,58 @@
 #include <dt-bindings/display/rk_fb.h>
 
 / {
+	vpu: vpu_service@ff650000 {
+		compatible = "rockchip,vpu_service";
+		rockchip,grf = <&grf>;
+		iommu_enabled = <1>;
+		reg = <0x0 0xff650000 0x0 0x800>;
+		interrupts = <GIC_SPI 113 IRQ_TYPE_LEVEL_HIGH>,
+			<GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "irq_dec", "irq_enc";
+		clocks = <&cru ACLK_VCODEC>, <&cru HCLK_VCODEC>;
+		clock-names = "aclk_vcodec", "hclk_vcodec";
+		resets = <&cru SRST_H_VCODEC>, <&cru SRST_A_VCODEC>;
+		reset-names = "video_h", "video_a";
+		name = "vpu_service";
+		dev_mode = <0>;
+		status = "disabled";
+	};
+
+	vpu_mmu: vpu_mmu {
+		dbgname = "vpu";
+		compatible = "rockchip,vpu_mmu";
+		reg = <0x0 0xff650800 0x0 0x40>;
+		interrupts = <GIC_SPI 115 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "vpu_mmu";
+		status = "disabled";
+	};
+
+	rkvdec: rkvdec@ff660000 {
+		compatible = "rockchip,rkvdec";
+		rockchip,grf = <&grf>;
+		iommu_enabled = <1>;
+		reg = <0x0 0xff660000 0x0 0x400>;
+		interrupts = <GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "irq_dec";
+		clocks = <&cru ACLK_VDU>, <&cru HCLK_VDU>,<&cru SCLK_VDU_CA>,<&cru SCLK_VDU_CORE>;
+		clock-names = "aclk_vcodec", "hclk_vcodec", "clk_cabac", "clk_core";
+		resets = <&cru SRST_H_VDU>, <&cru SRST_A_VDU>;
+		reset-names = "video_h", "video_a";
+		dev_mode = <2>;
+		name = "rkvdec";
+		status = "disabled";
+	};
+
+	vdec_mmu: vdec_mmu {
+		dbgname = "vdec";
+		compatible = "rockchip,vdec_mmu";
+		reg = <0x0 0xff660480 0x0 0x40>,
+		      <0x0 0xff6604c0 0x0 0x40>;
+		interrupts = <GIC_SPI 117 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "vdec_mmu";
+		status = "disabled";
+	};
+
 	fb: fb {
 		compatible = "rockchip,rk-fb";
 		rockchip,disp-mode = <DUAL>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-monkey.dts b/arch/arm64/boot/dts/rockchip/rk3399-monkey.dts
index 777df951bc3c..aeca00141ed1 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-monkey.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-monkey.dts
@@ -171,3 +171,18 @@
 	dr_mode = "peripheral";
 };
 
+&vpu {
+	status = "okay";
+};
+
+&rkvdec {
+	status = "okay";
+};
+
+&vpu_mmu {
+	status = "okay";
+};
+
+&vdec_mmu {
+	status = "okay";
+};
-- 
2.35.3

